pkg/tls/tls.go

```diff
@@ -72,12 +80,23 @@ func getCertificates(clientCertFile, clientKeyFile string) ([]tls.Certificate, e
 	return []tls.Certificate{}, nil
 }
 
-func readFile(path string) ([]byte, error) {
-	file, err := os.Open(path)
-	if err != nil {
-		return nil, err
+// readPEMBytes will parse the certificate on input and return the pem byte
+// content. It accepts a pem certificate or the file path to a pem certificate.
+func readPEMBytes(cert string) ([]byte, error) {
+	if isPEMString(cert) {
+		return []byte(cert), nil
 	}
-	defer file.Close()
 
-	return io.ReadAll(file)
+	return os.ReadFile(cert)
+}
+
+// isPEMString returns true if the provided string match a PEM formatted certificate.
+func isPEMString(s string) bool {
+	// trim the certificates to make sure we tolerate any yaml weirdness
+	trimmedStr := strings.TrimSpace(s)
```

Comment:
nit: You could reuse `s` here.

```suggestion
	s = strings.TrimSpace(s)
```